### PR TITLE
feat: show raw transaction traces

### LINKED ISSUE #1656: Feature request: transaction traces
Sometimes it would be useful to view (raw) transaction traces. Examples on other explorers:
* etherscan: https://etherscan.io/vmtrace?txhash=0x1c9679c0f6e1539075b5c52939ec72eef5995c69c571f9f84f599768cc479707&type=parity#raw
* etherchain-light: https://light.etherchain.org/tx/raw/0xb685caa9686474cbc9d6e577845cf617c9992c04e2bfca1fd3f8248460c1b365

### PR DESCRIPTION
**Pull request by zachdaniel**
Resolves #1656 

### Enhancements
* Adds code to derive the original raw traces from the internal transactions. There are some caveats:
  * ~We don't store the value for `subtraces`, and I'm unsure as of yet if it can be derived from what we have.~ ~This can be derived, coming shortly.~ done
  * We store some data as a Decimal, but it is only present in the trace as an integer, so if we end up storing data for an internal transaction as 1.5, it would show as `2` in the raw trace. Since its an integer in the API, I doubt we have to worry about it not being a round number though.

- [x] I need to write some tests to cover the various kinds of traces. That should not take long.

- [x] I also need to move the code to a module in explorer, instead of ethereum jsonrpc.

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so if necessary
raw-trace

### CODE REVIEW

**Comment by goodsoft** [comment]
@vbaranov seems to be the similar problem with `BLOCK_TRANSFORMER`.
Does it manifest on `master` and on current Sokol/Rinkeby deployment branch?

**Comment by vbaranov** [comment]
@goodsoft let me check my setup. Possibly I messed up with the set of env vars.

**Comment by zachdaniel** [comment]
I don’t think that this PR could relate to fetcher errors in any way.

**Comment by vbaranov** [comment]
Yeah, that's right. There was a problem with my setup. Raw traces look good to me 👍

File: apps/block_scout_web/lib/block_scout_web/views/transaction_raw_trace_view.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  def render("scripts.html", %{conn: conn}) do
+    render_scripts(conn, "raw_trace/code_highlighting.js")
+  end
+
+  def raw_traces_with_lines(internal_transactions) do
+    internal_transactions
+    |> InternalTransaction.internal_transactions_to_raw()
+    |> Jason.encode!(pretty: true)
+    |> String.split("\n")
+    |> Enum.with_index()
```
Comment (goodsoft) [inline apps/block_scout_web/lib/block_scout_web/views/transaction_raw_trace_view.ex]: I think, we'd rather number them from 1.

File: apps/explorer/lib/explorer/chain/internal_transaction.ex [hunk 2]
```diff
@@ -497,4 +497,121 @@ defmodule Explorer.Chain.InternalTransaction do
...
+      direct_descendant?(trace_address, trace_address_candidate)
+    end)
+  end
+
+  defp direct_descendant?([], [_]), do: true
+
+  defp direct_descendant?([elem | remaining_left], [elem | remaining_right]),
+    do: direct_descendant?(remaining_left, remaining_right)
+
+  defp direct_descendant?(_, _), do: false
```
Comment (goodsoft) [inline apps/explorer/lib/explorer/chain/internal_transaction.ex]: `List.starts_with?`?

Comment (zachdaniel) [reply apps/explorer/lib/explorer/chain/internal_transaction.ex]: There is also the requirement that there is only one more element in the list than in the parent list, so I figured I'd combine the checks. I could do `List.starts_with?(left, right) && length(left) == length(right) + 1` though, if this is too unclear.

Comment (zachdaniel) [reply apps/explorer/lib/explorer/chain/internal_transaction.ex]: If you look at the base case you can see that it checks that the right side only has one element remaining.

Comment (goodsoft) [reply apps/explorer/lib/explorer/chain/internal_transaction.ex]: Okay, I see. Current solution is fine, as it won't traverse the lists twice.

**Pull request merged**